ministryofjustice/analytics-platform-ops
# Add authentication proxy to RShiny apps

## What

* Updated helm chart for `shiny-app` to deploy a [generic auth-proxy](https://github.com/ministryofjustice/analytics-platform-auth-proxy) in front of it, to protect access to the app with an Auth0 OTP
* Modified `livenessProbe` to allow a few seconds for shiny app to start, to avoid flapping

## How to review

1. Create an RShiny app in a Github repository similar to [this one](https://github.com/moj-analytical-services/andy-test/blob/master/)
2. Deploy the app with Jenkins
3. Browse to the app URL (Jenkins should provide this on successful build)
4. Be prompted to enter your email address by Auth0
5. Receive a login link email
6. Click on the login link and be redirected to your shiny app

[Shiny app authentication and authorisation {8}](https://trello.com/c/uW6wEJvs)
